### PR TITLE
feat: `.add` to fix policy corruption

### PR DESCRIPTION
**Pull request by remy**
- [x] Ready for review
- [x] Follows CONTRIBUTING rules
- [x] Reviewed by @adrukh 

#### What does this PR do?

Handles policies that have lost a leading `-` in the array of paths. This happens in the Snyk CLI when an existing policy is updated with new rules.

Part of the change fixes these corrupted policies on the fly, but this change also introduces a `policy.add()` function so the new policy rules can be added properly without the merge failure.

Other changes include:

- Switch from node@5 to node@6 in travis
- Add validation and fix if the policy is corrupted (where the merge of an old policy and new fails to update properly)
- Add `.create` method to create an empty policy
- Move detection for early, pre-v1 policy format to the loading phase (rather than filter)
- Increase code coverage

#### Where should the reviewer start?

Core change is in `lib/parser/v1.js` where the sample policy file looks like this:

```text
ignore:
  'npm:hawk:20160119':
    - request > hawk:
        reason: no patch available
        expires: '2016-02-19T11:25:17.324Z'
      octonode > request > hawk:
        reason: None given
        expires: '2016-05-24T13:46:19.066Z'
    - npm > request > hawk:
        reason: no patch available
        expires: '2016-02-19T11:25:17.325Z'
```

#### How should this be manually tested?

Probably best to read the tests in `test/unit/policy-parse.test.js`.

#### Any background context you want to provide?

Corruption is happening here: https://github.com/Snyk/snyk/blob/378bbd194c7c6ad6786706f9bc765c86ac537dad/lib/protect/index.js#L28

#### What are the relevant tickets?

https://snyksec.atlassian.net/browse/SC-1106

### CODE REVIEW

File: lib/parser/v1.js [hunk 1]
```diff
@@ -9,5 +9,63 @@ module.exports = function imports(policy) {
     policy.patch = {};
   }
 
+  checkForOldFormat(policy.ignore);
+  validate(policy.ignore);
+  validate(policy.ignore);
```
Comment (adrukh) [inline lib/parser/v1.js]: Did you mean `validate(policy.patch);`?

File: package.json [hunk 2]
```diff
@@ -18,11 +18,12 @@
   "license": "Apache-2.0",
   "devDependencies": {
     "jscs": "^2.11.0",
+    "lodash": "^3.10.1",
```
Comment (adrukh) [inline package.json]: I hear 3.x is outdated, how about "^4.11.1"?

File: lib/parser/v1.js (hunk 1 above)
Comment (remy) [inline lib/parser/v1.js]: Yes, need to fix that immediately.

File: package.json (hunk 2 above)
Comment (remy) [inline package.json]: I intentionally added this to match snyk CLI, but iirc I don't use this in the tests anymore, so will remove in next push to this branch.

**Pull request merged**